tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -745,6 +745,12 @@ void SpirvEmitter::HandleTranslationUnit(ASTContext &context) {
   if (context.getDiagnostics().hasErrorOccurred())
     return;
 
+  if (spirvOptions.debugInfoRich) {
+    emitWarning("Debug information for methods won't be linked to their class. "
+                "See https://github.com/KhronosGroup/SPIRV-Registry/issues/203",
+                {});
+  }
```

Comment:
This gets output even if there are no member functions. Is that what the SPIR WG agreed to? Should this be moved to DebugTypeVisitor.cpp, and only emit the warning if there was a function that should be skipped?  However, we do not want to issue the warning multiple times.

Reply:
Yeah, originally I added the warning there, but the visitor is called multiple times, even with 1 single method. (At least if I put the warning where I know we have the debug instruction "badly" generated)

Reply:
What could be done is storing some bit in the visitor, like "have you encountered a member function", then read this bit once in the SpirvEmitter, but it looked a bit ugly

Reply:
@s-perron anything blocking on your end? Would you be Ok with this warning being shown once, but even when no class is compiled or do you want me to add some state to minimize printing?

Reply:
Thanks for the reminder. I'm fine with the warning as is. We'll see if it bothers people.  

I just thought of something. Can dxc turn warnings into errors? If so, what happens when this warning is issued?

Reply:
So there is `-WX`, and `-Werror` is accepted. Both looks to keep this warning as a warning. And there is also `-no-warnings` which disables this warning.